{"cells":[{"source":"<a href=\"https://www.kaggle.com/code/digitalbro/multimodal-sc-integration-meta-resources?scriptVersionId=104112207\" target=\"_blank\"><img align=\"left\" alt=\"Kaggle\" title=\"Open in Kaggle\" src=\"https://kaggle.com/static/images/open-in-kaggle.svg\"></a>","metadata":{},"cell_type":"markdown","outputs":[],"execution_count":0},{"cell_type":"markdown","id":"46b003cd","metadata":{"papermill":{"duration":0.004334,"end_time":"2022-08-24T16:07:40.431218","exception":false,"start_time":"2022-08-24T16:07:40.426884","status":"completed"},"tags":[]},"source":["The aim for this notebook is the create a central log (meta resource) of all the information and resources I've come across in the discussions, other notebooks, in my own discovery. I tend to find it best for me to keep track of all the many nooks one can store information on Kaggle in one place. \n","\n","The following notebook has tried to aggregate important papers / resources that will help in understanding how this one might go about completing this challenge. I've also provided where these resources have been taken at the bottom. "]},{"cell_type":"markdown","id":"ef965542","metadata":{"papermill":{"duration":0.002844,"end_time":"2022-08-24T16:07:40.437444","exception":false,"start_time":"2022-08-24T16:07:40.4346","status":"completed"},"tags":[]},"source":["## Experiments\n","### CITE-seq\n","![citeseq.com](https://citeseq.files.wordpress.com/2018/02/figure1.png?w=700)\n"," * Protein (CITE-seq): Enables leveraging legacy markers used over the last decades by immunologists to define cell spectrums \n"," \n"," ### ATAC-seq\n"," \n"," * ATACseq: Key to defining immune cell states transitional states best defined by up- and down- regulation of critical transcription factors (usually poorly captured transcriptionally) \n"," * RNAseq - Key to degining cell type/states indentity through modules of genes uniquely expressed\n"," "]},{"cell_type":"markdown","id":"c823b272","metadata":{"papermill":{"duration":0.002732,"end_time":"2022-08-24T16:07:40.443222","exception":false,"start_time":"2022-08-24T16:07:40.44049","status":"completed"},"tags":[]},"source":["# Dictionary \n","* ATAC - assay for transposase-accessible chromatin \n","* CITE-seq: Cellular indexing of transcriptomesand epitopes by sequencing\n"]},{"cell_type":"markdown","id":"2cb31005","metadata":{"papermill":{"duration":0.002881,"end_time":"2022-08-24T16:07:40.449077","exception":false,"start_time":"2022-08-24T16:07:40.446196","status":"completed"},"tags":[]},"source":["# Papers\n","* [Integrated analysis of multimodal single-cell data](https://www.sciencedirect.com/science/article/pii/S0092867421005833)\n","* [New horizons in the stormy sea of multimodal single-cell data integration](https://www.sciencedirect.com/science/article/abs/pii/S1097276521010741)\n","* [Computation principles and challenges in single-cell data integration](https://www.nature.com/articles/s41587-021-00895-7)\n","* [Diagonal integration of multimodal single-cell data: potential pitfalls and paths forward](https://www.nature.com/articles/s41467-022-31104-x)\n","* [Bi-order multimodal integration of single-cell data](https://genomebiology.biomedcentral.com/articles/10.1186/s13059-022-02679-x)\n","* [Multimodal single-cell approaches shed light on T cell heterogeneity](https://www.sciencedirect.com/science/article/pii/S0952791519300469)\n","* [Cobolt: integrative analysis of multimodal single-cell sequencing data](https://genomebiology.biomedcentral.com/articles/10.1186/s13059-021-02556-z)\n","* [Human haematopoietic stem cell lineage commitment is a continuous process](https://www.nature.com/articles/ncb3493)\n","* [Normalizing and denoising protein expression data from droplet-based single cell profiling](https://www.nature.com/articles/s41467-022-29356-8)\n","* [BABEL enables cross-modality translation between multiomic profiles at single-cell resolution](https://pubmed.ncbi.nlm.nih.gov/33827925/)\n","* [Current best practices in single-cell RNA-seq analysis: a tutorial](https://www.embopress.org/doi/full/10.15252/msb.20188746)\n","## Preprint\n","* [Computational challenges in cell cycle analysis using single cell transcriptomics](https://arxiv.org/abs/2208.05229) \n","* [Multimodal single-cell chromatin analysis with Signac](https://www.biorxiv.org/content/10.1101/2020.11.09.373613v1.abstract) \n","* [MultiVI: deep generative model for the integration of multimodal-data](https://www.biorxiv.org/content/10.1101/2021.08.20.457057v1)"]},{"cell_type":"markdown","id":"70130b88","metadata":{"papermill":{"duration":0.002745,"end_time":"2022-08-24T16:07:40.455072","exception":false,"start_time":"2022-08-24T16:07:40.452327","status":"completed"},"tags":[]},"source":["# Experimental Details\n","* Cell Lines Used: https://allcells.com/research-grade-tissue-products/mobilized-leukopak/\n","* Multignome - ATAC + Gene Expression: https://www.10xgenomics.com/products/single-cell-multiome-atac-plus-gene-expression\n","    * Chromatin accessibility to predict gene expression\n","* CITESeq - Single Cell Gene Expression: https://support.10xgenomics.com/permalink/getting-started-single-cell-gene-expression-with-feature-barcoding-technology\n","    * Cell Surface Reagent - https://www.biolegend.com/en-gb/products/totalseq-b-human-universal-cocktail-v1dot0-20960"]},{"cell_type":"markdown","id":"44d015d8","metadata":{"papermill":{"duration":0.002744,"end_time":"2022-08-24T16:07:40.460894","exception":false,"start_time":"2022-08-24T16:07:40.45815","status":"completed"},"tags":[]},"source":["# Additional Information \n","* [EBI Ensemble Id Information](https://www.ebi.ac.uk/training/online/courses/ensembl-browsing-genomes/navigating-ensembl/investigating-a-gene/#:~:text=Ensembl%20gene%20IDs%20begin%20with,of%20species%20other%20than%20human)\n","* [Eleven Grand Challenges in Single-Cell Data Science](https://genomebiology.biomedcentral.com/articles/10.1186/s13059-020-1926-6)"]},{"cell_type":"markdown","id":"e3b92995","metadata":{"papermill":{"duration":0.002772,"end_time":"2022-08-24T16:07:40.46665","exception":false,"start_time":"2022-08-24T16:07:40.463878","status":"completed"},"tags":[]},"source":["# Kaggle Notebooks\n","* [scRNA-seq ðŸ§¬: Differential Expression with scVI](https://www.kaggle.com/code/hiramcho/scrna-seq-differential-expression-with-scvi/notebook)\n","* [scRNA-seq ðŸ§¬: Scanpy & SCMER for Feature Selection](https://www.kaggle.com/code/hiramcho/scrna-seq-scanpy-scmer-for-feature-selection/notebook)\n","* [scRNA-seq ðŸ§¬: scGAE with Spektral and RAPIDS](https://www.kaggle.com/code/hiramcho/scrna-seq-scgae-with-spektral-and-rapids/notebook)\n","* [scATAC-seq ðŸ§¬: Feature Importance with TabNet](https://www.kaggle.com/code/hiramcho/scatac-seq-feature-importance-with-tabnet/notebook)\n","* [scATAC-seq ðŸ§¬: EpiScanpy & PeakVI](https://www.kaggle.com/code/hiramcho/scatac-seq-episcanpy-peakvi)\n"]},{"cell_type":"markdown","id":"0e4c6046","metadata":{"papermill":{"duration":0.002749,"end_time":"2022-08-24T16:07:40.472439","exception":false,"start_time":"2022-08-24T16:07:40.46969","status":"completed"},"tags":[]},"source":["# External Notebooks / Packages \n","* [KNN Solution](https://github.com/adavoudi/msci_knn)"]},{"cell_type":"markdown","id":"7a179e88","metadata":{"papermill":{"duration":0.002761,"end_time":"2022-08-24T16:07:40.478195","exception":false,"start_time":"2022-08-24T16:07:40.475434","status":"completed"},"tags":[]},"source":["# Learning Resources\n","* [MIA: Multimodal Single-cell data, open benchmarks, and a NeurIPS 2021](https://www.biolegend.com/en-gb/products/totalseq-b-human-universal-cocktail-v1dot0-20960) - *video* \n","* [Open Problems in Single Cells Analysis](https://openproblems.bio/neurips_docs/data/about_multimodal/)\n","    * Open problems in scAnalysis - "]},{"cell_type":"markdown","id":"d7fb842c","metadata":{"papermill":{"duration":0.002789,"end_time":"2022-08-24T16:07:40.483954","exception":false,"start_time":"2022-08-24T16:07:40.481165","status":"completed"},"tags":[]},"source":["# Potentially Useful Packages (Python)\n","* [muon](https://muon.readthedocs.io/en/latest/api/generated/muon.atac.pp.tfidf.html?highlight=tfidf) - muon is a Python framework for multimodal omics analysis. While there are many features that muon brings to the table, there are three key areas that its functionality is focused on.\n","* [scanpy](https://scanpy.readthedocs.io/en/stable/index.html) - Scanpy is a scalable toolkit for analyzing single-cell gene expression data built jointly with anndata. It includes preprocessing, visualization, clustering, trajectory inference and differential expression testing. The Python-based implementation efficiently deals with datasets of more than one million cells.\n","* [anndata](https://anndata.readthedocs.io/en/latest/#) - nndata is a Python package for handling annotated data matrices in memory and on disk, positioned between pandas and xarray. anndata offers a broad range of computationally efficient features including, among others, sparse data support, lazy operations, and a PyTorch interface.\n","* [Xarray](https://docs.xarray.dev/en/v0.9.2/dask.html) - xarray (formerly xray) is an open source project and Python package that aims to bring the labeled data power of pandas to the physical sciences, by providing N-dimensional variants of the core pandas data structures. **This will help split up the large dataset**\n","* [ivis](https://bering-ivis.readthedocs.io/en/latest/index.html) - ivis is a machine learning library for reducing dimensionality of very large datasets using Siamese Neural Networks. ivis preserves global data structures in a low-dimensional space, adds new data points to existing embeddings using a parametric mapping function, and scales linearly to millions of observations. The algorithm is described in detail in Structure-preserving visualisation of high dimensional single-cell datasets."]},{"cell_type":"markdown","id":"5c063269","metadata":{"papermill":{"duration":0.00291,"end_time":"2022-08-24T16:07:40.489905","exception":false,"start_time":"2022-08-24T16:07:40.486995","status":"completed"},"tags":[]},"source":["# Appendix \n","\n","## Cell Types \n","* MasP = Mast Cell Progenitor\n","* MkP = Megakaryocyte Progenitor\n","* NeuP = Neutrophil Progenitor\n","* MoP = Monocyte Progenitor\n","* EryP = Erythrocyte Progenitor\n","* HSC = Hematoploetic Stem Cell\n","* BP = B-Cell Progenitor"]},{"cell_type":"markdown","id":"5681e4d4","metadata":{"papermill":{"duration":0.002776,"end_time":"2022-08-24T16:07:40.495687","exception":false,"start_time":"2022-08-24T16:07:40.492911","status":"completed"},"tags":[]},"source":["# Thanks \n","At this point I just have aggregated the information from various notebooks and discussions as a way to keep track of all of the various notebooks \n","* Thomas Shelby - https://www.kaggle.com/competitions/open-problems-multimodal/discussion/344686\n","* Daniel Burkhardt - https://www.kaggle.com/competitions/open-problems-multimodal/discussion/344607\n","* Kaggle Data Details the team at Cellarity - https://www.kaggle.com/competitions/open-problems-multimodal/data\n","* Peter Holderrieth - https://www.kaggle.com/competitions/open-problems-multimodal/discussion/345958\n","* MarÃ­lia Prat - https://www.kaggle.com/competitions/open-problems-multimodal/discussion/346686\n","* Alireza - https://www.kaggle.com/competitions/open-problems-multimodal/discussion/346894"]},{"cell_type":"code","execution_count":null,"id":"ed88a71a","metadata":{"papermill":{"duration":0.002762,"end_time":"2022-08-24T16:07:40.501453","exception":false,"start_time":"2022-08-24T16:07:40.498691","status":"completed"},"tags":[]},"outputs":[],"source":[]}],"metadata":{"kernelspec":{"display_name":"Python 3","language":"python","name":"python3"},"language_info":{"codemirror_mode":{"name":"ipython","version":3},"file_extension":".py","mimetype":"text/x-python","name":"python","nbconvert_exporter":"python","pygments_lexer":"ipython3","version":"3.7.12"},"papermill":{"default_parameters":{},"duration":9.667951,"end_time":"2022-08-24T16:07:41.127229","environment_variables":{},"exception":null,"input_path":"__notebook__.ipynb","output_path":"__notebook__.ipynb","parameters":{},"start_time":"2022-08-24T16:07:31.459278","version":"2.3.4"}},"nbformat":4,"nbformat_minor":5}